lxd/network/network_utils.go

```diff
@@ -639,6 +639,109 @@ func randomSubnetV6() (string, error) {
 	return "", fmt.Errorf("Failed to automatically find an unused IPv6 subnet, manual configuration required")
 }
 
+// randomAddressInSubnet finds a random address within a given subnet. If given, the validate function is called for
+// each randomly generated value. Each time validate returns an error, another IP address will be generated until the
+// context is cancelled.
+func randomAddressInSubnet(ctx context.Context, ipnet *net.IPNet, validate func(net.IP) error) (net.IP, error) {
```

Comment:
Why pointer for net.IPNet here?

Reply:
I don't have a good reason! A `net.IPNet` is mutable anyway as the underlying type of `net.IP` is a byte slice, but I agree we should pass by value if there is no good reason to pass by reference. Will change.